File: Data Structures & Algorithms/majority-element/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        count = 0
        m = nums[0]

        for i in nums:
            if count ==0:
                m = i
            
            if m == i:
                count+=1
            else:
                count-=1
        
        return m
        

File: Data Structures & Algorithms/contains-duplicate-ii/submission-3.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashSet<Integer> s = new HashSet<>();
        int l = 0;

        for (int i = 0; i < nums.length; i++) {
            if (i - l > k) {
                s.remove(nums[l]);
                l++;
            }

            if (s.contains(nums[i])) {
                return true;
            }
            s.add(nums[i]);
        }
        return false;
    }
}